### PR TITLE
Remove the TransportKind enum from the SDK since it is not used.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1650

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]
well that was hard :)

**Comment by vhvb1989** [comment]
Just wait on merging until we merge the PR after releasing that updates the changelog please

**Review by RickWinter — APPROVED** [review]

**Pull request merged**
